Extension/lib/filter/rules/url-filter-rule.js

```diff
@@ -172,6 +181,35 @@ UrlFilterRule.prototype.isFiltered = function (requestUrl, thirdParty, requestCo
 };
 
 /**
+ * Checks if specified content type is suitable.
+ *
+ * @param contentType Request content type (UrlFilterRule.contentTypes)
+ */
+UrlFilterRule.prototype.checkContentType = function (contentType) {
+    var contentTypeMask = UrlFilterRule.contentTypes[contentType];
+    return this.checkContentTypeMask(contentTypeMask);
+};
+
+/**
+ * Checks if specified content type is suitable.
+ *
+ * @param contentType Request content type mask (UrlFilterRule.contentTypes)
+ */
+UrlFilterRule.prototype.checkContentTypeMask = function (contentTypeMask) {

```

Comment:
This method is used in checkContentType only.

You could simply merge this code there.